core/src/main/java/org/apache/calcite/sql/SqlSelect.java

```diff
@@ -83,14 +85,16 @@ public SqlSelect(SqlParserPos pos,
                    SqlNode offset,
                    SqlNode fetch,
                    SqlNodeList hints) {
-    this(pos, keywordList, /*topN=*/ null, selectList, from, where, groupBy,
-        having, qualify, windowDecls, orderBy, offset, fetch, hints);
+    this(pos, keywordList, /*topN=*/ null, /*exceptExpression=*/ null, selectList,
+        from, where, groupBy, having, qualify, windowDecls, orderBy, offset,
+        fetch, hints);
   }
 
   public SqlSelect(SqlParserPos pos,
```

Comment:
Let's add a new constructor. Otherwise, we could break downstream users outside the codebase.

Reply:
I'm a bit confused what should this constructor have as inputs?

Reply:
One model we have been following is to have a base constructor which takes all of the fields as input. Then a variety of derived constructors that take just certain fields and then call the base constructor with the omitted fields set to null.

My comment here was that we shouldn't modify an existing derived constructor because we could then break a caller that's currently using it. Instead, we should create a brand new one which adds the exceptExpression on top of other fields.

Reply:
I may be missing something but I feel like this is what I did? I may be wrong but it looks like the base constructor is the one I added the field `SqlNode exceptExpression` to. And then I passed null to this field in the other existing constructors.

Reply:
Actually nevermind, I think I see what you mean.

Reply:
That's true, but now you'd need to add a new derived constructor to handle the previous arg list (without exceptExpression), right?

IIUC, otherwise, any callers that were using the previous version of the base constructor will automatically break after this change.